lib/what/what.js

```diff
@@ -0,0 +1,200 @@
+'use babel'
+
+import {CompositeDisposable} from 'atom'
+import {utf8OffsetForBufferPosition} from './../utils'
+import {computeArgs, adjustPositionForGuru} from './../guru-utils'
+import os from 'os'
+
+class What {
+  constructor (goconfig) {
+    this.subscriptions = new CompositeDisposable()
+    this.goconfig = goconfig
+    this.running = false
+    this.subscriptions.add(atom.config.observe('go-plus.guru.highlightIdentifiers', (v) => {
+      this.shouldDecorate = v
+      if (this.shouldDecorate) {
+        this.subscribeToCursorEvents()
+      } else {
+        this.unsubscribeToCursorEvents()
+      }
+    }))
+  }
+
+  dispose () {
+    if (this.subscriptions) {
+      this.subscriptions.dispose()
+    }
+    this.subscriptions = null
+    this.unsubscribeToCursorEvents()
+    this.goconfig = null
+    this.running = false
+  }
+
+  subscribeToCursorEvents () {
+    this.cursorSubscriptions = new CompositeDisposable()
+    this.cursorSubscriptions.add(atom.workspace.observeTextEditors((editor) => {
+      if (!editor || !editor.getBuffer()) {
+        return
+      }
+
+      const editorSubscriptions = new CompositeDisposable()
+      editorSubscriptions.add(editor.onDidChangeCursorPosition(({cursor, newBufferPosition}) => {
+        this.run(editor, newBufferPosition, cursor, this.shouldDecorate)
+      }))
+      editorSubscriptions.add(editor.onDidDestroy(() => {
+        editorSubscriptions.dispose()
+        this.currentEditorLayer = null
+      }))
+      this.cursorSubscriptions.add(editorSubscriptions)
+    }))
+  }
+
+  unsubscribeToCursorEvents () {
+    this.clearMarkers()
+    if (this.cursorSubscriptions) {
+      this.cursorSubscriptions.dispose()
+      this.cursorSubscriptions = null
+    }
+  }
+
+  // parseStream (jsonStream) {
```

Comment:
Yes, it should probably be in guru-utils, and remove the logic about he index and instead return an array of stuff from index 0.